dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -881,6 +886,49 @@ public List<User> updateUserMetadata(@ApiParam(hidden = true) @Parameter(hidden
         return userDAO.findAll();
     }
 
+    @POST
+    @UnitOfWork
+    @RolesAllowed("admin")
+    @Path("/updateUserWorkflows")
+    @Operation(operationId = "checkWorkflowOwnership", description = "Check workflow ownership", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponse(responseCode = HttpStatus.SC_NO_CONTENT
+        + "", description = "Successfully updated workflow ownership for all users", content = @Content(array = @ArraySchema(schema = @Schema(implementation = User.class))))
+    @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + "", description = HttpStatusMessageConstants.FORBIDDEN)
+    public Response updateUserWorkflows(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User adminUser) {
+        final List<Long> allGitHubUsers = userDAO.findAllGitHubUserIds();
+        allGitHubUsers.forEach(userId -> {
+            // Break into many transactions so all workflows are not loaded into memory
+            new TransactionHelper(sessionFactory).transaction(() -> {
+                tokenDAO.findGithubByUserId(userId).stream().limit(1).forEach(gitHubToken -> {
+                    try {
+                        final SourceCodeRepoInterface sourceCodeRepo =
+                            SourceCodeRepoFactory.createSourceCodeRepo(gitHubToken);
+                        final Set<String> orgMemberships = sourceCodeRepo.getOrganizationMemberships();
+                        final Set<GitRepo> gitRepos = sourceCodeRepo.getRepoLevelAccessRepositories();
+                        final User user = userDAO.findById(userId);
+                        user.getEntries().stream().filter(Workflow.class::isInstance)
+                            .map(Workflow.class::cast)
+                            .filter(workflow -> {
+                                final boolean userHasOrganizationAccess =
+                                    orgMemberships.contains(workflow.getOrganization());
+                                return !userHasOrganizationAccess && !hasRepoLevelAccess(gitRepos, workflow);
+                            })
+                            .forEach(workflow -> workflow.removeUser(user));
+                    } catch (Exception e) {
+                        // LOG and continue -- don't let one user's invalid token block everybody
```

Comment:
Does it matter if each transaction is rolled back if an exception is thrown?  Probably not, but if so, this catch should be moved to the outside of the `TransactionHelper.transaction` call (which rolls the transaction back if an exception propagates out of the lambda).

Reply:
Good catch; the tests were failing with an invalid token and this fixed it, but it would go ahead and try to commit the changes, which we wouldn't want.